components/patina_smbios/src/component.rs

```diff
@@ -55,78 +59,87 @@ impl Default for SmbiosProviderManager {
 }
 
 impl SmbiosProviderManager {
-    /// Create a new SMBIOS provider manager with default SMBIOS 3.9 version
+    /// Create a new SMBIOS provider manager
     pub fn new() -> Self {
-        Self { manager: SmbiosManager::new(3, 9) }
+        Self {}
     }
 
     /// Initialize the SMBIOS provider and register it as a service
     fn entry_point(
-        mut self,
+        self,
         config: Option<Config<SmbiosConfiguration>>,
         mut commands: Commands,
         boot_services: StandardBootServices,
     ) -> Result<()> {
-        log::trace!("Initializing SMBIOS Provider...");
-
         let cfg = config.map(|c| (*c).clone()).unwrap_or_default();
 
-        // Update manager with configured version
-        self.manager = SmbiosManager::new(cfg.major_version, cfg.minor_version);
+        // Create the manager with configured version
+        let manager = SmbiosManager::new(cfg.major_version, cfg.minor_version);
 
-        log::trace!("SMBIOS version {}.{}", cfg.major_version, cfg.minor_version);
-
-        // Install the C protocol for EDKII compatibility
-        match install_smbios_protocol(&self.manager, &boot_services) {
+        // Install the protocol - this transfers ownership to the global singleton
+        match install_smbios_protocol(manager, &boot_services) {
             Ok(handle) => {
-                log::info!("SMBIOS C protocol installed successfully at handle {:?}", handle);
+                log::info!("SMBIOS protocol installed successfully at handle {:?}", handle);
             }
             Err(e) => {
-                log::warn!("Failed to install SMBIOS C protocol: {:?}", e);
-                // Continue anyway - the Rust service will still work
+                log::error!("Failed to install SMBIOS protocol: {:?}", e);
+                // Cannot proceed without the manager - this is a fatal error
+                panic!("SMBIOS manager installation failed");
```

Comment:
I don't think we should panic here. debug_assert and bail is reasonable, but in production we'd want to be able to continue if possible.